### PR TITLE
Escape asterisks when generating markdown doc files

### LINKED ISSUE #5219: 🐛 Asterisk should be escaped in generated markdown files
🐛 This is a bug in markdown generation. The asterisks should be escaped. I'll file a follow-up issue on that.

_Originally posted by @Youssef1313 in https://github.com/dotnet/roslyn-analyzers/pull/5210#r665961235_

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #5219

FYI @RussKie

### CODE REVIEW

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 1]
```diff
@@ -413,6 +412,16 @@ void createAnalyzerDocumentationFile()
                 }
             }
 
+            string escapeMarkdown(string text)
+            {
+                // Escape generic arguments to ensure they are not considered as HTML elements
+                text = Regex.Replace(text, "(<.+?>)", "\\$1");
+
+                // Escape asterisks.
+                text = text.Replace("*", @"\*");
```
Comment (Youssef1313) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:421]: I kept it simple and escaped asterisks only.

**Review by RussKie — APPROVED** [review]
:shipit:

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 2]
```diff
@@ -413,6 +412,16 @@ void createAnalyzerDocumentationFile()
                 }
             }
 
+            string escapeMarkdown(string text)
```
Comment (Evangelink) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:415]: Could be made `static`.

Comment (Evangelink) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:415]: Probably better to stick with pascal case.

Comment (Evangelink) [reply src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:415]: Ah nevermind the other functions are with camel case...

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 3]
```diff
@@ -413,6 +412,16 @@ void createAnalyzerDocumentationFile()
...
+            {
+                // Escape generic arguments to ensure they are not considered as HTML elements
+                text = Regex.Replace(text, "(<.+?>)", "\\$1");
```
Comment (Evangelink) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:418]: It's a personal taste but you could rewrite the function just as a chain of function calls:

```csharp
 string escapeMarkdown(string text) =>
	return text
		// Escape generic arguments to ensure they are not considered as HTML elements
		.Replace("(<.+?>)", "\\$1")
		// Escape asterisks.
        .Replace("*", @"\*");
```

**Review by jmarolf — APPROVED** [review]

**Pull request merged**